server/game/cards/07-GR/SoulExchange.js

```diff
@@ -0,0 +1,56 @@
+const Card = require('../../Card.js');
+
+function firstCreatureInDiscard(context) {
```

Comment:
Can you rewrite like this to avoid the double filter repetition?

```
function firstCreatureInDiscard(context) {
    const player = !context.select || context.select === 'Mine'
        ? context.player : context.player.opponent;
    return player.discard.find((c) => c.type === 'creature')
```

Reply:
Done, thanks.